R/utilities-mpi-error-handling.R

```diff
@@ -0,0 +1,153 @@
+#' @title checkSimulationLoaded
+#' @description Check that the simulation has been loaded on the core.
+#' @param simulation the simulation object to be checked if loaded successfully
+#' @param simulationFilePath the path to the simulation object to be checked if loaded successfully
+#' @export
+checkSimulationLoaded <- function(simulation, simulationFilePath) {
+  success <- FALSE
+  if (identical(simulation$sourceFile, simulationFilePath)) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+#' @title checkLibraryLoaded
+#' @description Check that a library has been successfully loaded on the core.
+#' @param libraryName the library to be checked if loaded successfully
+#' @export
+checkLibraryLoaded <- function(libraryName) {
+  success <- FALSE
+  if (libraryName %in% .packages()) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+
+#' @title checkNotNull
+#' @description Check that an object is not null.
+#' @param object to be checked if not null
+#' @export
+checkNotNull <- function(object) {
+  success <- FALSE
+  if (!is.null(object)) {
```

Comment:
Why do we need a method to replace a core method?

Reply:
Removed and replaced with !is.null()